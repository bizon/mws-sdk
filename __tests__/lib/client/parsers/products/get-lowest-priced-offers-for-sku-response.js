const parseXml = require('../../../../../lib/client/parsers')
const parseGetLowestPricedOffersForSkuResponse = require('../../../../../lib/client/parsers/products/get-lowest-priced-offers-for-sku-response')

describe('lib.client.parsers.products.get-lowest-priced-offers-for-sku-response', () => {
  it('should parse the GetLowestPricedOffersForSKUResponse success example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetLowestPricedOffersForSKUResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <GetLowestPricedOffersForSKUResult MarketplaceID="ATVPDKIKX0DER" SKU="GE Product" ItemCondition="New"
          status="Success">
          <Identifier>
            <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
            <SellerSKU>GE Product</SellerSKU>
            <ItemCondition>New</ItemCondition>
            <TimeOfOfferChange>2015-07-19T23:15:11.859Z</TimeOfOfferChange>
          </Identifier>
          <Summary>
            <TotalOfferCount>1</TotalOfferCount>
            <NumberOfOffers>
              <OfferCount condition="new" fulfillmentChannel="Amazon">1</OfferCount>
            </NumberOfOffers>
            <LowestPrices>
              <LowestPrice condition="new" fulfillmentChannel="Amazon">
                <LandedPrice>
                  <CurrencyCode>USD</CurrencyCode>
                  <Amount>32.99</Amount>
                </LandedPrice>
                <ListingPrice>
                  <CurrencyCode>USD</CurrencyCode>
                  <Amount>32.99</Amount>
                </ListingPrice>
                <Shipping>
                  <CurrencyCode>USD</CurrencyCode>
                  <Amount>0.00</Amount>
                </Shipping>
              </LowestPrice>
            </LowestPrices>
            <BuyBoxPrices>
              <BuyBoxPrice condition="New">
                <LandedPrice>
                  <CurrencyCode>USD</CurrencyCode>
                  <Amount>32.99</Amount>
                </LandedPrice>
                <ListingPrice>
                  <CurrencyCode>USD</CurrencyCode>
                  <Amount>32.99</Amount>
                </ListingPrice>
                <Shipping>
                  <CurrencyCode>USD</CurrencyCode>
                  <Amount>0.00</Amount>
                </Shipping>
              </BuyBoxPrice>
            </BuyBoxPrices>
            <ListPrice>
              <CurrencyCode>USD</CurrencyCode>
              <Amount>58.34</Amount>
            </ListPrice>
            <SuggestedLowerPricePlusShipping>
              <CurrencyCode>USD</CurrencyCode>
              <Amount>32.99</Amount>
            </SuggestedLowerPricePlusShipping>
            <BuyBoxEligibleOffers>
              <OfferCount condition="new" fulfillmentChannel="Amazon">1</OfferCount>
            </BuyBoxEligibleOffers>
          </Summary>
          <Offers>
            <Offer>
              <MyOffer>false</MyOffer>
              <SubCondition>new</SubCondition>
              <SellerFeedbackRating>
                <SellerPositiveFeedbackRating>100.0</SellerPositiveFeedbackRating>
                <FeedbackCount>1</FeedbackCount>
              </SellerFeedbackRating>
              <ShippingTime minimumHours="0" maximumHours="0" availabilityType="NOW" />
              <ListingPrice>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>32.99</Amount>
              </ListingPrice>
              <Shipping>
                <CurrencyCode>USD</CurrencyCode>
                <Amount>0.00</Amount>
              </Shipping>
              <IsFulfilledByAmazon>true</IsFulfilledByAmazon>
              <IsBuyBoxWinner>true</IsBuyBoxWinner>
              <IsFeaturedMerchant>true</IsFeaturedMerchant>
            </Offer>
          </Offers>
        </GetLowestPricedOffersForSKUResult>
        <ResponseMetadata>
          <RequestId>fd2e6c6d-0b6d-499b-9fea-074175c0547a</RequestId>
        </ResponseMetadata>
      </GetLowestPricedOffersForSKUResponse>`
    )

    const res = parseGetLowestPricedOffersForSkuResponse(
      '/products:GetLowestPricedOffersForSKUResponse',
      doc
    )

    expect(res).toMatchSnapshot()
  })

  it('should parse the GetLowestPricedOffersForSKUResponse success with no offers example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetLowestPricedOffersForSKUResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <GetLowestPricedOffersForSKUResult MarketplaceID="ATVPDKIKX0DER" SKU="0I-RIAS-6UA0" ItemCondition="new"
          status="NoBuyableOffers">
          <Identifier>
            <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
            <SellerSKU>0I-RIAS-6UA0</SellerSKU>
            <ItemCondition>New</ItemCondition>
          </Identifier>
          <Summary>
            <TotalOfferCount>0</TotalOfferCount>
          </Summary>
          <Offers />
        </GetLowestPricedOffersForSKUResult>
        <ResponseMetadata>
          <RequestId>75621aa7-9c8b-40be-9bf9-3ac1efdcdb87</RequestId>
        </ResponseMetadata>
      </GetLowestPricedOffersForSKUResponse>`
    )

    const res = parseGetLowestPricedOffersForSkuResponse(
      '/products:GetLowestPricedOffersForSKUResponse',
      doc
    )

    expect(res).toMatchSnapshot()
  })

  it('should parse the GetLowestPricedOffersForSKUResponse missing shipping charge example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0" encoding="UTF-8"?>
      <GetLowestPricedOffersForSKUResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <GetLowestPricedOffersForSKUResult MarketplaceID="ATVPDKIKX0DER" SKU="0I-RIAS-6UA0" ItemCondition="new"
          status="NoOfferDueToMissingShippingCharge">
          <Identifier>
            <MarketplaceId>ATVPDKIKX0DER</MarketplaceId>
            <SellerSKU>0I-RIAS-6UA0</SellerSKU>
            <ItemCondition>New</ItemCondition>
          </Identifier>
          <Summary>
            <TotalOfferCount>0</TotalOfferCount>
          </Summary>
          <Offers />
        </GetLowestPricedOffersForSKUResult>
        <ResponseMetadata>
          <RequestId>75621aa7-9c8b-40be-9bf9-3ac1efdcdb87</RequestId>
        </ResponseMetadata>
      </GetLowestPricedOffersForSKUResponse>`
    )

    const res = parseGetLowestPricedOffersForSkuResponse(
      '/products:GetLowestPricedOffersForSKUResponse',
      doc
    )

    expect(res).toMatchSnapshot()
  })

  it('should parse the GetLowestPricedOffersForSKUResponse too soon for processing example response from MWS doc', () => {
    const doc = parseXml(
      `<?xml version="1.0"?>
      <GetLowestPricedOffersForSKUResponse xmlns="http://mws.amazonservices.com/schema/Products/2011-10-01">
        <GetLowestPricedOffersForSKUResult MarketplaceID="A2EUQ1WTGCTBG2" SKU="ACA-8039-6PK-DL57" ItemCondition="New"
          status="ActiveButTooSoonForProcessing">
          <Identifier>
            <MarketplaceId>A2EUQ1WTGCTBG2</MarketplaceId>
            <SellerSKU>ACA-8039-6PK-DL57</SellerSKU>
            <ItemCondition>New</ItemCondition>
          </Identifier>
          <Summary>
            <TotalOfferCount>0</TotalOfferCount>
            <OffersAvailableTime>2017-03-08T07:13:21.763Z</OffersAvailableTime>
          </Summary>
          <Offers />
        </GetLowestPricedOffersForSKUResult>
        <ResponseMetadata>
          <RequestId>1212334</RequestId>
        </ResponseMetadata>
      </GetLowestPricedOffersForSKUResponse>`
    )

    const res = parseGetLowestPricedOffersForSkuResponse(
      '/products:GetLowestPricedOffersForSKUResponse',
      doc
    )

    expect(res).toMatchSnapshot()
  })
})
