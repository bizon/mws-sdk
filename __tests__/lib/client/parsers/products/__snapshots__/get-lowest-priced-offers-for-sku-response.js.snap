// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib.client.parsers.products.get-lowest-priced-offers-for-sku-response should parse the GetLowestPricedOffersForSKUResponse missing shipping charge example response from MWS doc 1`] = `
Object {
  "getLowestPricedOffersForSkuResult": Object {
    "identifier": Object {
      "asin": null,
      "itemCondition": "New",
      "marketplaceId": "ATVPDKIKX0DER",
      "sellerSku": "0I-RIAS-6UA0",
      "timeOfOfferChange": null,
    },
    "itemCondition": "new",
    "marketplaceId": "ATVPDKIKX0DER",
    "offers": Array [],
    "sku": "0I-RIAS-6UA0",
    "status": "NoOfferDueToMissingShippingCharge",
    "summary": Object {
      "buyBoxEligibleOffers": Array [],
      "buyBoxPrices": Array [],
      "competitivePriceThreshold": null,
      "listPrice": null,
      "lowestPrices": Array [],
      "numberOfOffers": Array [],
      "offersAvailableTime": null,
      "salesRankings": Array [],
      "suggestedLowerPricePlusShipping": null,
      "totalOfferCount": 0,
    },
  },
  "responseMetadata": Object {
    "requestId": "75621aa7-9c8b-40be-9bf9-3ac1efdcdb87",
  },
}
`;

exports[`lib.client.parsers.products.get-lowest-priced-offers-for-sku-response should parse the GetLowestPricedOffersForSKUResponse success example response from MWS doc (updated) 1`] = `
Object {
  "getLowestPricedOffersForSkuResult": Object {
    "identifier": Object {
      "asin": null,
      "itemCondition": "New",
      "marketplaceId": "ATVPDKIKX0DER",
      "sellerSku": "GE Product",
      "timeOfOfferChange": 2015-07-19T23:15:11.859Z,
    },
    "itemCondition": "New",
    "marketplaceId": "ATVPDKIKX0DER",
    "offers": Array [
      Object {
        "conditionNotes": "Some notes",
        "isBuyBoxWinner": true,
        "isFeaturedMerchant": true,
        "isFulfilledByAmazon": true,
        "listingPrice": Object {
          "amount": 32.99,
          "currencyCode": "USD",
        },
        "myOffer": false,
        "points": null,
        "primeInformation": Object {
          "isNationalPrime": true,
          "isPrime": true,
        },
        "sellerFeedbackRating": Object {
          "feedbackCount": 1,
          "sellerPositiveFeedbackRating": 100,
        },
        "sellerId": "SELLERID",
        "shipping": Object {
          "amount": 0,
          "currencyCode": "USD",
        },
        "shippingTime": Object {
          "availabilityType": "NOW",
          "availableDate": null,
          "maximumHours": 0,
          "minimumHours": 0,
        },
        "shipsFrom": null,
        "subCondition": "new",
      },
    ],
    "sku": "GE Product",
    "status": "Success",
    "summary": Object {
      "buyBoxEligibleOffers": Array [
        Object {
          "condition": "new",
          "count": 1,
          "fulfillmentChannel": "Amazon",
        },
      ],
      "buyBoxPrices": Array [
        Object {
          "condition": "New",
          "landedPrice": Object {
            "amount": 32.99,
            "currencyCode": "USD",
          },
          "listingPrice": Object {
            "amount": 32.99,
            "currencyCode": "USD",
          },
          "points": null,
          "shipping": Object {
            "amount": 0,
            "currencyCode": "USD",
          },
        },
      ],
      "competitivePriceThreshold": null,
      "listPrice": Object {
        "amount": 58.34,
        "currencyCode": "USD",
      },
      "lowestPrices": Array [
        Object {
          "condition": "new",
          "fulfillmentChannel": "Amazon",
          "landedPrice": Object {
            "amount": 32.99,
            "currencyCode": "USD",
          },
          "listingPrice": Object {
            "amount": 32.99,
            "currencyCode": "USD",
          },
          "points": null,
          "shipping": Object {
            "amount": 0,
            "currencyCode": "USD",
          },
        },
      ],
      "numberOfOffers": Array [
        Object {
          "condition": "new",
          "count": 1,
          "fulfillmentChannel": "Amazon",
        },
      ],
      "offersAvailableTime": null,
      "salesRankings": Array [
        Object {
          "productCategoryId": "book_display_on_website",
          "rank": 401,
        },
        Object {
          "productCategoryId": "15356791",
          "rank": 5,
        },
        Object {
          "productCategoryId": "3153",
          "rank": 8,
        },
        Object {
          "productCategoryId": "17468",
          "rank": 16,
        },
      ],
      "suggestedLowerPricePlusShipping": Object {
        "amount": 32.99,
        "currencyCode": "USD",
      },
      "totalOfferCount": 1,
    },
  },
  "responseMetadata": Object {
    "requestId": "fd2e6c6d-0b6d-499b-9fea-074175c0547a",
  },
}
`;

exports[`lib.client.parsers.products.get-lowest-priced-offers-for-sku-response should parse the GetLowestPricedOffersForSKUResponse success with no offers example response from MWS doc 1`] = `
Object {
  "getLowestPricedOffersForSkuResult": Object {
    "identifier": Object {
      "asin": null,
      "itemCondition": "New",
      "marketplaceId": "ATVPDKIKX0DER",
      "sellerSku": "0I-RIAS-6UA0",
      "timeOfOfferChange": null,
    },
    "itemCondition": "new",
    "marketplaceId": "ATVPDKIKX0DER",
    "offers": Array [],
    "sku": "0I-RIAS-6UA0",
    "status": "NoBuyableOffers",
    "summary": Object {
      "buyBoxEligibleOffers": Array [],
      "buyBoxPrices": Array [],
      "competitivePriceThreshold": null,
      "listPrice": null,
      "lowestPrices": Array [],
      "numberOfOffers": Array [],
      "offersAvailableTime": null,
      "salesRankings": Array [],
      "suggestedLowerPricePlusShipping": null,
      "totalOfferCount": 0,
    },
  },
  "responseMetadata": Object {
    "requestId": "75621aa7-9c8b-40be-9bf9-3ac1efdcdb87",
  },
}
`;

exports[`lib.client.parsers.products.get-lowest-priced-offers-for-sku-response should parse the GetLowestPricedOffersForSKUResponse too soon for processing example response from MWS doc 1`] = `
Object {
  "getLowestPricedOffersForSkuResult": Object {
    "identifier": Object {
      "asin": null,
      "itemCondition": "New",
      "marketplaceId": "A2EUQ1WTGCTBG2",
      "sellerSku": "ACA-8039-6PK-DL57",
      "timeOfOfferChange": null,
    },
    "itemCondition": "New",
    "marketplaceId": "A2EUQ1WTGCTBG2",
    "offers": Array [],
    "sku": "ACA-8039-6PK-DL57",
    "status": "ActiveButTooSoonForProcessing",
    "summary": Object {
      "buyBoxEligibleOffers": Array [],
      "buyBoxPrices": Array [],
      "competitivePriceThreshold": null,
      "listPrice": null,
      "lowestPrices": Array [],
      "numberOfOffers": Array [],
      "offersAvailableTime": 2017-03-08T07:13:21.763Z,
      "salesRankings": Array [],
      "suggestedLowerPricePlusShipping": null,
      "totalOfferCount": 0,
    },
  },
  "responseMetadata": Object {
    "requestId": "1212334",
  },
}
`;
